### PR TITLE
Update dogs3 dependency

### PR DESCRIPTION
**Pull request by frankmcsherry**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by antiguru — APPROVED** [review]
Looks great, thanks!

File: src/compute/src/render/join/delta_join.rs [hunk 1]
```diff
@@ -461,7 +461,9 @@ where
         let (oks, errs2) = dogsdogsdogs::operators::half_join::half_join_internal_unsafe(
             &updates,
             trace,
-            |time| time.step_back(),
+            |time, antichain| {
+                antichain.insert(time.step_back());
```
Comment (petrosagg) [inline src/compute/src/render/join/delta_join.rs:465]: This doesn't yet give us "precise" compaction as described by the description of [this PR](https://github.com/TimelyDataflow/differential-dataflow/pull/457) yet, right? Do we plan to do this in MZ?

Comment (frankmcsherry) [reply src/compute/src/render/join/delta_join.rs:465]: It does, it turns out. MZ's use is limited to `Product<u64, PointStamp<u64>>`, or whatever, and that happens to be a [distributive lattice](https://en.wikipedia.org/wiki/Distributive_lattice) for which compaction by the meet of the frontier equals compaction by the frontier.

Comment (petrosagg) [reply src/compute/src/render/join/delta_join.rs:465]: Very interesting, thanks for the explanation!

**Pull request merged**